source/extensions/filters/network/http_connection_manager/config.cc

```diff
@@ -296,7 +298,33 @@ HttpConnectionManagerConfig::HttpConnectionManagerConfig(
     final_rid_config.mutable_typed_config()->PackFrom(
         envoy::extensions::request_id::uuid::v3::UuidRequestIdConfig());
   }
-  request_id_extension_ = Http::RequestIDExtensionFactory::fromProto(final_rid_config, context_);
+  request_id_extension_ =
+      Envoy::Http::RequestIDExtensionFactory::fromProto(final_rid_config, context_);
+
+  // Check if IP detection extensions were configured.
```

Comment:
If you want to avoid all the namespace changes in this file and have it be a bit cleaner you can do the following:
1) Make the xff extension an extension that is forced into the build. See `_required_extensions` in all_extensions.bzl
2) Instead of manually adding the extension here, check if the extensions are empty, if so push on a new config that contains just an empty config with the right type URL text for the XFF extension, then go through the normal loading code.

This will not pollute the namespace here, and also unify the loading code without any special casing.

Up to you if you want to do ^ or not.

Reply:
The verbosity isn't that terrible all things considered, so leaving it this way for now.